### PR TITLE
fix(themes): fix theme flash on initial page render

### PR DESCRIPTION
**Pull request by florkbr**
### Description

This change prevents the UI from flashing the default light theme before the saved dark theme is applied. It adds a shared, build-time-generated inline theme bootstrap that runs before PatternFly CSS, applies the saved dark preference, and applies the forced felt/glass themes for `/lightwell`.

The dark-mode initialization also preserves the pre-paint theme while feature flags are loading, avoiding an intermediate light-theme render.

[RHCLOUD-XXXXX](https://issues.redhat.com/browse/RHCLOUD-XXXXX) (TBD)

---

#### Before:

https://github.com/user-attachments/assets/1b47cc4f-6edf-4464-80e1-be3aab6beeae

#### After:

https://github.com/user-attachments/assets/f2c1b7d3-0d83-47f9-ae8a-6e72c4e79a6f

---

### Anything reviewers should know?

- `config/theme-bootstrap.js` is loaded by webpack in Node and stringified into an inline blocking script. The generated function runs in the browser before PatternFly CSS loads.
- Both `index.ejs` and `indexRes.ejs` use the same bootstrap implementation.
- `THEME_STORAGE_KEY` is defined in the existing `src/utils/consts.ts` storage-key module.
- The Playwright test keeps a local key constant because Playwright specs run in Node, while importing `src/utils/consts.ts` evaluates browser-only `location` logic.
- The explicit Lightwell route rule is intentional. Lightwell is a Chrome-owned route, so Chrome knows its felt/glass theme before React mounts.
- Frontend Operator configuration is currently exposed through `chrome-service` after `index.html` loads, so it cannot prevent the initial CSS paint. Supporting tenant-configurable pre-paint themes in a future S3/CloudFront deployment would require static build-time metadata, edge/server-side HTML injection, or hiding the UI until configuration loads.
- Unit coverage validates both HTML templates, Lightwell routes, dark/felt/glass combinations, and unavailable localStorage.
- The bootstrap cannot know feature-flag state before the page loads. If dark mode is later disabled while `chrome:theme` is still set to `dark`, the UI may briefly paint dark before `useTheme` removes the stale class and applies light mode.
- We intentionally do not clear `chrome:theme`, so the user’s preference is preserved if the feature is enabled again.
- Eliminating this edge case requires paint gating until flags resolve or server/edge-provided flag state, which is outside the scope of this change.
- Verification completed:
  - `npm test -- --runInBand`: 1,174 tests passed
  - `npm run build`: passed
  - `npm run lint:js -- --quiet`: passed
  - `npm run lint`: existing unrelated Prettier issues remain under `build-tools/`
  - `npm run build:res`: existing `pf5styles` template-parameter issue remains

---

### Checklist

- [x] Accessibility: color contrast, keyboard nav, screen reader tested (or N/A)
- [x] All PR checks pass locally (build, lint, test)
- [x] No unrelated changes included
- [x] QE: Unit coverage added for theme bootstrap and flag-loading behavior
- [ ] UX: end-user UX modified, designs need sign-off

### AI disclosure

Assisted by: OpenCode

## New Features
- Added early theme initialization to reduce flashes of incorrect styling during page load.
- Lightwell routes now automatically apply the appropriate Lightwell visual themes.
- Saved dark mode and system color-scheme preferences are respected during startup.

## Bug Fixes
- Improved theme behavior while feature settings load, including consistent persistence across views and sessions.
- Dark mode now correctly reflects the page’s initial theme state, including when storage is unavailable.

### CODE REVIEW

**Comment by florkbr** [comment]
@coderabbitai full review

File: src/indexRes.ejs [hunk 1]
```diff
@@ -12,6 +12,9 @@
     <meta http-equiv="x-ua-compatible" content="ie=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
     <link rel="icon" type="image/png" href="https://access.redhat.com/webassets/avalon/g/favicon.ico">
+    <script type="text/javascript">
+        <%= themeBootstrap %>
```
[the remark this answers is not in the document]
Comment (florkbr) [reply src/indexRes.ejs:16]: @coderabbitai this is outside the scope of this change (albeit a valid concern). I will take it back to my team for review and follow up in a different PR if that is ok.

[the remark this answers is not in the document]
Comment (florkbr) [reply src/indexRes.ejs:16]: No - I will open a JIRA in our backlog. Thanks

[the remark this answers is not in the document]
Comment (florkbr) [reply src/indexRes.ejs:16]: @coderabbitai No - I will open a JIRA in our backlog. Thanks

[the remark this answers is not in the document]
Comment (florkbr) [reply src/indexRes.ejs:16]: @coderabbitai created: https://redhat.atlassian.net/browse/RHCLOUD-51326

**Review by justinorringer — APPROVED** [review]
Looking good!

**Review by catastrophe-brandon — APPROVED** [review]
